stdlib/REPL/src/LineEdit.jl

```diff
@@ -2829,16 +2829,20 @@ function prompt!(term::TextTerminal, prompt::ModalInterface, s::MIState = init_s
         activate(prompt, s, term, term)
         old_state = mode(s)
         l = Base.ReentrantLock()
-        t = @async while true
-            fcn = take!(s.async_channel)
-            status = @lock l fcn(s)
+        t = Threads.@spawn while true
+            wait(s.async_channel)
+            status = @lock l begin
+                fcn = take!(s.async_channel)
```

Comment:
If the actual `take!`ing isn't happening on the main task, then the main task needs to terminate this task while holding the lock before returning.

Reply:
You mean something like close the channel on the main task and check after wait?
```
        t = Threads.@spawn while true
            wait(s.async_channel)
            isopen(s.async_channel) && break
            status = @lock l begin
                fcn = take!(s.async_channel)
```

Reply:
But then again, if every call to `prompt!` is creating this task, then we should tidy up before returning from `prompt!`, so it should close and reopen `s.async_channel` but that seems racy.

Reply:
Pushed a fix